scalameta/semanticdb-scalac-core/src/main/scala/scala/meta/internal/semanticdb/SymbolOps.scala

```diff
@@ -1,84 +1,96 @@
 package scala.meta.internal
 package semanticdb
 
+import java.util.HashMap
 import scala.{meta => m}
 import scala.meta.internal.inputs._
 
 trait SymbolOps { self: DatabaseOps =>
 
+  lazy val symbolCache = new HashMap[g.Symbol, m.Symbol]
   implicit class XtensionGSymbolMSymbol(sym: g.Symbol) {
     def toSemantic: m.Symbol = {
```

Comment:
Can we keep the original function unchanged except named `toSemanticUncached` and then create new `toSemantic` that uses caching? I feel it's difficult to follow the logic when the nesting is too deep. The big diff also makes it harder to review this change.

Reply:
Nevermind, I want to edit this method to fix #1194